lib/logic.rb

```diff
@@ -0,0 +1,106 @@
+class Game
+  attr_accessor :real_grid
+  def initialize()
+    @real_grid = Array.new(3, 0) { Array.new(3, 0) }
+    p @real_grid[0][1]
+    k = 1
+    3.times do |i|
+      3.times do |j|
+        @real_grid[i][j] = k.to_s
+        k += 1
+      end
+    end
+  end
+
+  def check_winner(board)
+    get_realG(board)
+    return 1 if horizontal(@real_grid, "X") || vertical(@real_grid,"X") || diagonal(@real_grid,"X")
```

Comment:
since you have `attr_reader :real_grid` then you shouldn't be calling `@real_grid` as such in here but `real_grid`.

Reply:
you are still calling @real_grid when I have specified in my previous review to just call it real_grid